src/components/table/table-column-resize/table-column-resize-expand/resizable-table-expand.directive.ts

```diff
@@ -0,0 +1,102 @@
+import { fromEvent } from 'rxjs';
+import { isPlatformBrowser } from '@angular/common';
+
+@Directive({
+    selector: '[uxResizableTableExpand]',
+    exportAs: 'ux-resizable-table-expand',
+    providers: [
+        {
+            provide: RESIZEABLE_TABLE_SERVICE_TOKEN,
+            useClass: ResizableTableExpandService
+        }
+    ],
+    host: {
+        'class': 'ux-resizable-table-expand',
+        '[class.ux-resizable-table-expand-overflow]': '_overflowX'
+    }
+})
+export class ResizableTableExpandDirective extends BaseResizableTableDirective implements AfterViewInit, OnDestroy {
+
+    /** Get all the column headers */
+    @ContentChildren(ResizableTableColumnComponent, { descendants: true }) columns: QueryList<ResizableTableColumnComponent>;
+
+    /** Has horizontal overflow */
+    _overflowX: boolean = false;
+
+    constructor(elementRef: ElementRef<HTMLTableElement>, @Inject(RESIZEABLE_TABLE_SERVICE_TOKEN) table: ResizableTableExpandService, renderer: Renderer2, resize: ResizeService, @Inject(PLATFORM_ID) private _platformId: Object) {
+        super(elementRef, table, renderer, resize);
+    }
+
+    ngAfterViewInit(): void {
+        super.ngAfterViewInit();
+
+        if (isPlatformBrowser(this._platformId)) {
+
+            const tableHeaders = this._elementRef.nativeElement.querySelectorAll('thead > tr');
+
+            for (const body of Array.from(this._elementRef.nativeElement.tBodies)) {
+                fromEvent(body, 'scroll').pipe(takeUntil(this._onDestroy)).subscribe(() => {
+                    Array.from(tableHeaders).forEach(thead => this._renderer.setStyle(thead, 'margin-left', `-${body.scrollLeft}px`));
+                });
+            }
+
+            /** checks if the table is resizing and allows for a class to be added for when moving from
+             overflow to no overflow */
+            this._table.onResize$.pipe(takeUntil(this._onDestroy)).subscribe(() => {
+                this._overflowX = this._elementRef.nativeElement.tBodies[0].scrollWidth > this._elementRef.nativeElement.tBodies[0].offsetWidth;
+            });
+
+        }
+
+    }
+
+    ngOnDestroy(): void {
+        /** destroying watching for resizing */
+        this._onDestroy.next();
+        this._onDestroy.complete();
+    }
+
+    /**
+     * If this is being used within a modal the table width may initially be zero. This can cause some issues when it does actually appear
+     * visibily on screen. We should only setup the table once we actually have a width/
+     */
+    onTableReady(): void {
+
+        // if we have already initialised or the table width is currently 0 then do nothing
+        if (this._initialised || this.getScrollWidth() === 0) {
+            return;
+        }
+
+        // ensure we initially set the table width
+        this._table.tableWidth = this.getScrollWidth();
+
+        // set the columns - prevent expression changed error
+        requestAnimationFrame(() => {
```

Comment:
I notice if I hit reset then start switching between the standard fixed header and expand fixed header modes there is a bit of flickering going on. While this is not a big deal as its unlikely an application would allow switching between the two modes, we can fix this easily. Can we swap all the uses of `requestAnimationFrame` in both directives with : `Promise.resolve().then(() => ...)`.

Promise resolve is a Microtask which gets executed before a requestAnimationFrame so this should reduce any flickering

Reply:
Works alot smoother now! Thanks!